### PR TITLE
feat(py/plugins/google-genai): Deep Research, Antigravity, and Lyria interaction models

### PR DESCRIPTION
**Pull request by cabljac**
Third slice of the #5806 split: Deep Research background model with cancel support, Antigravity and Lyria foreground models on the Interactions transport, and a registry of known models with capability metadata. Rewrites lyria.py from the legacy Vertex-only LyriaModel. Model-level tests included here; plugin wiring and its integration tests land in the next slice.

Part of the stack splitting #5806; recombined it reproduces that PR byte-for-byte. Co-authored with @huangjeff5.

### Breaking changes

- `LyriaModel` is removed from `genkit_google_genai.models.lyria`; Lyria is now served by the Interactions-backed implementation (`create_lyria_action`). The package-level exports `LyriaConfig` and `LyriaVersion` are unchanged.

### CODE REVIEW

File: py/packages/genkit-google-genai/src/genkit_google_genai/models/deep_research.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    if config.google_search:
+        tools.append({'type': 'google_search'})
+    if config.url_context:
+        tools.append({'type': 'url_context'})
+    if config.code_execution:
+        tools.append({'type': 'code_execution'})
+    if config.file_search is not None:
+        tools.append({'type': 'file_search', **config.file_search.model_dump(exclude_none=True)})
+    for mcp_server in config.mcp_servers or []:
+        tools.append({'type': 'mcp_server', **mcp_server.model_dump(exclude_none=True)})
```
[the remark this answers is not in the document]
Comment (cabljac) [reply py/packages/genkit-google-genai/src/genkit_google_genai/models/deep_research.py:152]: The suggestion would break these requests rather than fix them. This body goes over a raw HTTP client verbatim, and the whole create payload is snake_case (agent_config, previous_interaction_id, user_input, mime_type in the content blocks), so camelCasing only these two nested blocks would produce a mixed-casing body. The SDK's own FileSearchDict/McpServerDict types also take snake_case keys. The alias_generator on these models handles the input direction (camelCased request config), not serialization.

File: py/packages/genkit-google-genai/src/genkit_google_genai/models/deep_research.py [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+        info=info,
+        config_schema=DeepResearchConfig,
+    )
+
+
+def build_tools(request: ModelRequest[DeepResearchConfig], config: DeepResearchConfig) -> list[dict[str, Any]]:
+    """Build Interactions API tool configurations for Deep Research."""
+    tools: list[dict[str, Any]] = []
+    if request.tools:
+        tools.extend(dict(to_interaction_tool(tool_def)) for tool_def in request.tools)
```
[the remark this answers is not in the document]
Comment (cabljac) [reply py/packages/genkit-google-genai/src/genkit_google_genai/models/deep_research.py:141]: Correct that the copy is a no-op (to_interaction_tool builds a fresh dict per call). No functional difference; will drop it in the next tidy commit on this slice.

File: py/packages/genkit-google-genai/src/genkit_google_genai/models/antigravity.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            )
+        require_interaction_steps(steps)
+        create_kwargs: dict[str, Any] = {
+            'agent': version,
+            'input': steps,
+            **create_options,
+            **passthrough,
+        }
+        # Default missing environment to remote; the API rejects unsupported values.
+        create_kwargs.setdefault('environment', DEFAULT_ENVIRONMENT)
```
[the remark this answers is not in the document]
Comment (cabljac) [reply py/packages/genkit-google-genai/src/genkit_google_genai/models/antigravity.py:119]: Nothing mutates the environment dict downstream today (the body goes straight to JSON serialization), but the shared-reference risk is real and the fix is cheap. Will take this.

File: py/packages/genkit-google-genai/src/genkit_google_genai/models/deep_research.py (hunk 1 above)
Comment (hilariie) [inline py/packages/genkit-google-genai/src/genkit_google_genai/models/deep_research.py:152]: Could we add tests for the `file_search` and `mcp_servers` branches here, plus `response_format_from_request`? None of the three are exercised in interactions_models_test.py.

These are the only places where the wire keys depend on `model_dump` being called without `by_alias`. Both config models set `alias_generator=to_camel`, so the correct `file_search_store_names` and `allowed_tools` only happen by omission. If someone adds `by_alias=True` later, it silently sends camelCase and no test fails.

File: py/packages/genkit-google-genai/src/genkit_google_genai/models/deep_research.py [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+    # Per-request options arrive camelCased (apiKey). Without accepting those
+    # names, the override silently disappears before check/cancel can reuse it.
+    model_config = ConfigDict(extra='allow', populate_by_name=True, alias_generator=to_camel)
+    api_key: str | None = None
+    base_url: str | None = None
+    api_version: str | None = None
+    # Milliseconds — applied to the HTTP call, not the create body.
+    timeout: float | None = None
+    custom_headers: dict[str, str] | None = None
+    thinking_summaries: Literal['auto', 'none'] | None = None
```
Comment (hilariie) [inline py/packages/genkit-google-genai/src/genkit_google_genai/models/deep_research.py:109]: The JS config accepts `thinkingSummaries: 'AUTO' | 'NONE'` and lowercases before building the request body. This accepts only `'auto' | 'none'`, and the same split applies to `visualization` and to `response_modalities` across all three models.

This means that a config copied from the JS samples can raise a ValidationError in Python.

File: py/packages/genkit-google-genai/src/genkit_google_genai/models/lyria.py [hunk 5]
```diff
@@ -14,188 +14,103 @@
...
+        modalities = create_options.get('response_modalities') or ['audio', 'text']
+        system_instruction, turns = split_system_instruction(request.messages or [])
+        steps = to_interaction_steps(ensure_tool_ids(turns))
+        create_kwargs: dict[str, Any] = {
+            'model': version,
+            'input': steps,
+            'response_modalities': modalities,
+            **passthrough,
+        }
+        if system_instruction:
```
Comment (hilariie) [inline py/packages/genkit-google-genai/src/genkit_google_genai/models/lyria.py:97]: This path has no JS counterpart. JS Lyria passes messages straight to `toInteractionSteps` with no system handling, so sending `system_instruction` and skipping the empty input guard is new behavior here. If this divergence from JS is intentional, there should be a test for the system-only case to pin the intent, since it is the one path that can reach the API with empty input.

File: py/packages/genkit-google-genai/src/genkit_google_genai/models/deep_research.py [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+    info = KNOWN_DEEP_RESEARCH_MODELS.get(clean) or DEEP_RESEARCH_INFO
+    # Prefer a version-specific label over the shared catalog default.
+    if info.label is None or info.label == DEEP_RESEARCH_INFO.label:
+        info = ModelInfo(label=f'Google AI - {clean}', supports=info.supports)
+    return model_ref(
+        name=clean,
+        namespace='googleai',
+        info=info,
+        config_schema=DeepResearchConfig,
+    )
```
Comment (hilariie) [inline py/packages/genkit-google-genai/src/genkit_google_genai/models/deep_research.py:134]: This reproduces the labeling that `deep_research_model_info` already does. Every entry in `KNOWN_DEEP_RESEARCH_MODELS` has either `label=None` or `label == DEEP_RESEARCH_INFO.label`, so the condition on line 127 is always true and the result is the same `ModelInfo` the registry helper returns.

Could this just call `deep_research_model_info(version)`? It keeps the labeling rule in the one file the registry docstring points people at.

**Review by hilariie — COMMENTED** [review]
The breaking changes section says `LyriaConfig` and `LyriaVersion` are unchanged, but both changed substantially. `LyriaVersion` swaps `LYRIA_002` for `LYRIA_3_CLIP` and `LYRIA_3_PRO`, and `LyriaConfig` replaces all four of its fields (`negative_prompt`, `seed`, `sample_count`, `location`) with the Interactions client options. Only the exported names are the same.

Worth noting that neither was reachable through any plugin before this stack, so nothing functional regresses, but the release notes get built from this section.
